### PR TITLE
Added documentation using sphinx with rtd theme

### PR DESCRIPTION
**Pull request by AvaisP**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,8 @@ node_modules
 
 # Webstorm specific files
 .idea
+
+# Docs
+_build
+_static
+_templates
```
Comment (deshraj) [inline .gitignore]: Please add a newline here.

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (deshraj) [inline docs/Makefile]: Please add a newline here.

Comment (AvaisP) [inline docs/Makefile]: The make file is generated by sphinx-quickstart on its own. Should I still edit it?

File: .gitignore (hunk 1 above)
Comment (AvaisP) [inline .gitignore]: At the end of the file right?

File: docs/Makefile (hunk 2 above)
Comment (deshraj) [inline docs/Makefile]: Oh yeah, you are right. You can still edit it. it's good to have a newline at the end of the file.

File: docs/Makefile [hunk 3]
```diff
@@ -17,4 +17,5 @@ help:
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
-	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
\ No newline at end of file
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
```
Comment (deshraj) [inline docs/Makefile]: It seems like there is some indentation in the last line. Can you please remove it?

**Pull request merged**